redisinsight/ui/src/components/base/utils/resize-observer/ResizeObserver.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useRef } from 'react'
+
+interface ResizeObserverProps {
+  children: (ref: React.Ref<HTMLDivElement>) => React.ReactNode
+  onResize: (dimensions: { height: number; width: number }) => void
+}
+
+export const ResizeObserver: React.FC<ResizeObserverProps> = ({
+  children,
+  onResize,
+}) => {
+  const containerRef = useRef<HTMLDivElement | null>(null)
+
+  useEffect(() => {
+    const element = containerRef.current
+    if (!element) return
+
+    const observer = new window.ResizeObserver(([entry]) => {
+      const { width, height } = entry.contentRect
+      onResize({ width, height })
+    })
+
+    observer.observe(element)
+
+    // eslint-disable-next-line consistent-return
+    return () => {
+      observer.disconnect()
+    }
+  }, [onResize])
```

Comment:
Probably depending on when the ref has an element in it won't hurt
```suggestion
  }, [onResize, containerRef.current])
```